### PR TITLE
Increase the number of morph targets to 1024 instead of 256

### PR DESCRIPTION
**Pull request by Obg1**
I'm using the fact that we are passing float4 so I can pass 4x data for the morph weights

### CODE REVIEW

**Comment by Obg1** [comment]
https://github.com/google/filament/issues/6142

**Comment by romainguy** [comment]
Please read the comments next to the files you modified. We cannot guarantee this change will work on ES 3.0

**Comment by romainguy** [comment]
Specifically in EngineEnums.h

**Comment by Obg1** [comment]
@romainguy I've reverted it to back 256, but since it's 256xfloat4 I'm allowing to use 256x4 = 1024 morph target.
Let me know if I've made a mistake as it's my first time making a change in Filament code

**Comment by pixelflinger** [comment]
Unfortunately this won't work on all implementations of GLES which has a limit to 256 layers for texture arrays.

**Comment by Obg1** [comment]
@pixelflinger I'm still using 256 layers but using the entire float4 instead of the float[0]. Look at the code changes

File: shaders/src/getters.vs [hunk 1]
```diff
@@ -100,7 +100,9 @@ void morphPosition(inout vec4 p) {
     ivec3 texcoord = ivec3(getVertexIndex() % MAX_MORPH_TARGET_BUFFER_WIDTH, getVertexIndex() / MAX_MORPH_TARGET_BUFFER_WIDTH, 0);
     uint c = getObjectUniforms().morphTargetCount;
     for (uint i = 0u; i < c; ++i) {
-        float w = morphingUniforms.weights[i][0];
+        uint pos = i / 4;
+        uint idx_pos = i % 4;
+        float w = morphingUniforms.weights[pos][idx_pos];
         if (w != 0.0) {
             texcoord.z = int(i);
```
Comment (pixelflinger) [inline shaders/src/getters.vs:107]: This is using more than 256 layers.

**Review by pixelflinger — CHANGES_REQUESTED** [review]

**Comment by pixelflinger** [comment]
> @pixelflinger I'm still using 256 layers but using the entire float4 instead of the float[0]. Look at the code changes

No, you are. I commented on the line in the shader that will access more than 256 layers.

File: shaders/src/getters.vs (hunk 1 above)
Comment (Obg1) [reply shaders/src/getters.vs:107]: pos is devided by 4, so 1024 / 4 is 256 maximum

**Comment by Obg1** [comment]
pos is devided by 4, so 1024 / 4 is 256 maximum

**Comment by bejado** [comment]
> pos is devided by 4, so 1024 / 4 is 256 maximum

The problematic line is:
```
texcoord.z = int(i);
```

**Comment by Obg1** [comment]
Closed as I'm still accessing more than 256 layers

**Pull request closed without merging**